### PR TITLE
Fixing up URL handling in Backup-DbaDatabase

### PR DESCRIPTION
**Pull request by Stuart-Moore**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [x] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Was getting bugged by having BackupDirectory and AzureBaseUrl Parameters, which made it different from Restore and Copy
And we didn't have azure striping nicely handled

### Approach
Can now pass URLs in via BackupDirectory
We'll stripe across multiple files on a URL or we'll strip across multiple URLs (if we're given a block blob (SAS auth))

IF someone uses AzureBaseUrl raise a warning saying it'll be disappearing as a parameter in v1.0

### CODE REVIEW

File: functions/Backup-DbaDatabase.ps1 [hunk 1]
```diff
@@ -101,7 +101,11 @@ function Backup-DbaDatabase {
 
     .PARAMETER AzureCredential
         The name of the credential on the SQL instance that can write to the AzureBaseUrl, only needed if using Storage access keys
+<<<<<<< HEAD
+        If using SAS credentials, the command will look for a credential with a name matching the AzureBaseUrl. Setting this parameter will set number of files to 1 as striped sets are not supported
+=======
         If using SAS credentials, the command will look for a credential with a name matching the AzureBaseUrl. As page blobs are used with this option we force the number of files to 1 and ignore any value passed in for BlockSize or MaxTransferSize
+>>>>>>> development
```
Comment (wsmelton) [inline functions/Backup-DbaDatabase.ps1]: Seems merge conflict headers got committed into the file....had this happen with VS Code before.

File: functions/Backup-DbaDatabase.ps1 [hunk 2]
```diff
@@ -263,8 +266,20 @@ function Backup-DbaDatabase {
                 $Filecount = $BackupDirectory.Count
             }
 
+            $urlcount = 0
+            Foreach ($directory in $BackupDirectory) {
```
Comment (wsmelton) [inline functions/Backup-DbaDatabase.ps1]: Lowercase `foreach` :pray:

File: functions/Backup-DbaDatabase.ps1 [hunk 3]
```diff
@@ -279,10 +294,19 @@ function Backup-DbaDatabase {
                     return
                 }
             }
-            if ($null -ne $AzureBaseUrl) {
-                $AzureBaseUrl = $AzureBaseUrl.Trim("/")
+            if ('' -ne $AzureBaseUrl) {
```
Comment (wsmelton) [inline functions/Backup-DbaDatabase.ps1]: A cleaner way to do this (and also check for null:

`if ( [string]::IsNullOrEmpty($AzureBaseUrl) ) {...`

File: functions/Backup-DbaDatabase.ps1 [hunk 4]
```diff
@@ -599,8 +623,8 @@ function Backup-DbaDatabase {
                                 }
                             }
                             $HeaderInfo | Add-Member -Type NoteProperty -Name BackupComplete -Value $BackupComplete
-                            $HeaderInfo | Add-Member -Type NoteProperty -Name BackupFile -Value (Split-Path $FinalBackupPath -Leaf)
-                            $HeaderInfo | Add-Member -Type NoteProperty -Name BackupFilesCount -Value $FinalBackupPath.Count
+                            $HeaderInfo | Add-Member -Type NoteProperty -Name backupFile -Value (Split-Path $FinalBackupPath -Leaf)
+                            $HeaderInfo | Add-Member -Type NoteProperty -Name backupFilesCount -Value $FinalBackupPath.Count
```
Comment (wsmelton) [inline functions/Backup-DbaDatabase.ps1]: Properties should be capital all the way through `BackupFilesCount`

File: functions/Backup-DbaDatabase.ps1 (hunk 1 above)
Comment (Stuart-Moore) [reply functions/Backup-DbaDatabase.ps1]: Yep, it's broken a load of stuff elsewhere (including casing as you point out below).

Will revert the merge and then sort everything else out.

**Comment by Stuart-Moore** [comment]
withdrawing this for now as don't have time to unpick the merge disaster atm

**Pull request closed without merging**